cmd/podman/shared/intermediate_varlink.go

```diff
@@ -0,0 +1,395 @@
+}
+
+// MakeVarlink creates a varlink transportable struct from GenericCLIResults
+func (g GenericCLIResults) MakeVarlink() iopodman.Create {
+	v := iopodman.Create{
+		Args:                   iopodman.CreateStringSlice{Value: g.InputArgs, Cr: iopodman.CreateResult{Name: "args", Changed: true}},
+		AddHost:                g.Find("add-host").ToVarlink().(iopodman.CreateStringSlice),
+		Annotation:             g.Find("annotation").ToVarlink().(iopodman.CreateStringSlice),
+		Attach:                 g.Find("attach").ToVarlink().(iopodman.CreateStringSlice),
+		BlkioWeight:            g.Find("blkio-weight").ToVarlink().(iopodman.CreateString),
+		BlkioWeightDevice:      g.Find("blkio-weight-device").ToVarlink().(iopodman.CreateStringSlice),
+		CapAdd:                 g.Find("cap-add").ToVarlink().(iopodman.CreateStringSlice),
+		CapDrop:                g.Find("cap-drop").ToVarlink().(iopodman.CreateStringSlice),
+		CgroupParent:           g.Find("cgroup-parent").ToVarlink().(iopodman.CreateString),
+		CidFile:                g.Find("cidfile").ToVarlink().(iopodman.CreateString),
+		ConmonPidfile:          g.Find("conmon-pidfile").ToVarlink().(iopodman.CreateString),
+		CpuPeriod:              g.Find("cpu-period").ToVarlink().(iopodman.CreateInt),
+		CpuQuota:               g.Find("cpu-quota").ToVarlink().(iopodman.CreateInt),
+		CpuRtPeriod:            g.Find("cpu-rt-period").ToVarlink().(iopodman.CreateInt),
+		CpuRtRuntime:           g.Find("cpu-rt-runtime").ToVarlink().(iopodman.CreateInt),
+		CpuShares:              g.Find("cpu-shares").ToVarlink().(iopodman.CreateInt),
+		Cpus:                   g.Find("cpus").ToVarlink().(iopodman.CreateFloat),
+		CpuSetCpus:             g.Find("cpuset-cpus").ToVarlink().(iopodman.CreateString),
+		CpuSetMems:             g.Find("cpuset-mems").ToVarlink().(iopodman.CreateString),
+		Detach:                 g.Find("detach").ToVarlink().(iopodman.CreateBool),
+		DetachKeys:             g.Find("detach-keys").ToVarlink().(iopodman.CreateString),
+		Device:                 g.Find("device").ToVarlink().(iopodman.CreateStringSlice),
+		DeviceReadBps:          g.Find("device-read-bps").ToVarlink().(iopodman.CreateStringSlice),
+		DeviceReadIops:         g.Find("device-read-iops").ToVarlink().(iopodman.CreateStringSlice),
+		DeviceWriteBps:         g.Find("device-write-bps").ToVarlink().(iopodman.CreateStringSlice),
+		DeviceWriteIops:        g.Find("device-write-iops").ToVarlink().(iopodman.CreateStringSlice),
+		Dns:                    g.Find("dns").ToVarlink().(iopodman.CreateStringSlice),
+		DnsOpt:                 g.Find("dns-opt").ToVarlink().(iopodman.CreateStringSlice),
+		DnsSearch:              g.Find("dns-search").ToVarlink().(iopodman.CreateStringSlice),
+		Entrypoint:             g.Find("entrypoint").ToVarlink().(iopodman.CreateString),
+		Env:                    g.Find("env").ToVarlink().(iopodman.CreateStringSlice),
+		EnvFile:                g.Find("env-file").ToVarlink().(iopodman.CreateStringSlice),
```

Comment:
Since we are removing EnvFile from being pushed.  Should we fix that before this is merged?